elastic/elastic-agent
# make policy change handler try all fleet hosts before failing

## What does this PR do?

Fix how the agent handles fleet-server hosts.

**It changes the fleet-server client to:**
 - when creating a new client:
   - succeed if at least one host is health
   - shuffle the hosts, avoiding all the agents reacing to the same fleet-server on the first request
 - makes `(remote.*Client).Send` try all the hosts before failing, returning the last error if all hosts fail
   - if debug logs are enabled, `Send` will log each error as with debug level
 - modifies `remopte.requestClient`:
   - now `requestClient` holds its host
   - remove `requestFunc`
   - `(remopte.requestClient).newRequest uses the new `host` property to build the final URL for the request

## Why is it important?

Right now, when creating a new fleet API client it only verifies if one new host is reachable, therefore if the first isn't reachable, it ignore the others.

## Checklist





## How to test this PR locally

Check https://github.com/elastic/elastic-agent/issues/1328 to see how to reproduce the issue.

## Related issues

- Fixes #1328
